lib/models/command.js

```diff
@@ -197,7 +198,7 @@ let Command = CoreObject.extend({
       return this._currentTask.onInterrupt();
     } else {
       // interrupt all external commands which don't use `runTask()` with an error
-      process.exit(1);
+      return RSVP.reject();
```

Comment:
> yes. If we do RSVP.reject(1) then capture-exit just outputs 1 to the console and exits with exitCode=1. As far as I understand this catch is designed to handle cleanup exceptions. Seems like it's better not to change this behavior.

Yes although the actual rejection reason is ignored; if we `RSVP.reject(4)` the exit code is still 1.

> @hjdivad what do you think? Do you have any objections about `resolve(desiredExitCode)`?

The problem is that a) you can have multiple handlers and b) the exit trigger can itself be a call to `process.exit` that passes in its own exit code.  The current implementation of capture-exit is meant to mimic node's behaviour as much as possible.

If you `process.exit(a)` and then  `process.exit(b)` again within a `process.on('exit')` handler, the exit code will be `a`; what happens in the `on('exit')` handlers doesn't affect the exit code.

---

Having the `command.onInterrupt` promise resolve to the desired exit code sounds good to me, but I don't particularly want to push this into capture-exit, which can already be a bit tricky to reason through.  I think I'd rather see the `onInterrupt` promise not added as an `onExit` handler but instead run within `will-interrupt-process` followed by a post-interrupt call (which by default would just `process.exit` but could avoid doing this in unit tests).

I'm not sure if this is the kind of thing @rwjblue had in mind in [this comment](https://github.com/ember-cli/ember-cli/issues/6779#issuecomment-280941092) or if he wanted exit handlers to more directly control the exit code.  @rwjblue what say you?